tests/e2e/local/kind/setup_test.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Remove old images.
+docker images 10.10.0.2:5000/*:latest -q | xargs docker rmi
+
+# Make and Push images to insecure local registry on VM.
+# Set GOOS=linux to make sure linux binaries are built on MacOS
+cd "$ISTIO/istio" || exit
+GOOS=linux make docker HUB=10.10.0.2:5000 TAG=latest
+GOOS=linux make push HUB=10.10.0.2:5000 TAG=latest
```

Comment:
Do we need to bring up a local registry? We removed it in the minikube case and I am not seeing that set up in this PR either... so this step had failed for me.

The push refers to repository [10.10.0.2:5000/pilot]
Get https://10.10.0.2:5000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

real	0m15.079s
user	0m0.092s
sys	0m0.046s
tools/istio-docker.mk:237: recipe for target 'push.docker.pilot' failed
make: *** [push.docker.pilot] Error 1



And then running e2e tests also fails.

2018-10-10T21:25:11.923280Z	error	Unable to create namespace istio-system: command failed: "Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kubernetes\")\n" exit status 1
2018-10-10T21:25:11.923301Z	error	Failed to deploy Istio.
2018-10-10T21:25:11.923319Z	error	Failed to complete Init. Error command failed: "Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"kubernetes\")\n" exit status 1